src/__rt/mod.rs

```diff
@@ -39,8 +42,26 @@ pub fn diverge<T>() -> T { loop {} }
 
 type StaticContext = *mut task::Context<'static>;
 
+#[cfg(feature = "std")]
 thread_local!(static CTX: Cell<StaticContext> = Cell::new(ptr::null_mut()));
 
+#[cfg(not(feature = "std"))]
```

Comment:
Can you explain this impl here? It seems totally unsafe to me-- `no_std` systems can still have threads, so accesses to `CTX` would be unsound, right?

Reply:
Yeah, I was kinda hoping someone else might have some idea how to make this safe on threaded `no_std`. The best solution I see is using generator arguments to avoid `CTX` entirely with and without `no_std`, but I don’t know what the current status on those are.

I guess there’s the old design of `futures 0.1` where you have to call an init routine to setup the thread locals on `no_std`, pushing the safety requirements onto the user of the library.

Reply:
After glancing at the `futures 0.1` `task::init` setup, I really don't want to have to fully understand that to make sure it's safe here. It's also really not how we would want to have the library work in the end since `futures-await` should be a transparent library able to be used by dependencies without the application needing to know about it.